sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java

```diff
@@ -34,9 +37,16 @@
     public static final String PRIMARY_KEY = "oid";
 
     /**
-     * Selection (WHERE clause) pattern for primary key search.
+     * Allowed multiple for maximum sizes.
      */
-    private static final String PRIMARY_KEY_SELECTION = "oid = ?";
+    @VisibleForTesting
+    static final int ALLOWED_SIZE_MULTIPLE = 4096;
+
+    /**
+     * Maximum number of entries of in memory database.
+     */
+    @VisibleForTesting
+    static final long IN_MEMORY_MAX_SIZE = 300;
```

Comment:
Should we keep this one for IMDB? It is okay but we probably need to consider removing IMDB. Just an opinion. 😄

Reply:
IMDB is used if there is a SQLite catastrophic error, so we just decided to keep the same logic as we do right now instead of writing a bunch more logic to ensure sizing... Although I guess that means that we might go over the manual limit? But we also don't really have any control over that if SQLite fails...

Reply:
We might need to revisit this if we're seeing that people run into this.  The problem is that we could break our contract on storage size if the 300 logs to be stored in the in-memory DB (which is only used in case of a SQLite failure/access error/etc.) go over the storage size.  It's a lot of extra logic for a pretty extreme edge case.

Reply:
@jaelim-ms We decide to keep the same 300 size limit logic for in memory database for simplicity. It is tricky to calculate the actual storage size for in memory data, because you will have to somehow compute the size by writing the data in the database, which is an expensive operation. It is a good enough for the purpose of a fail back solution. In case we find a better solution, we can always improve it later.